### PR TITLE
docs: warn about the typescript parser being a singleton

### LINKED ISSUE #422: ESLint Flat Configuration
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.36.0

### What version of `eslint-plugin-svelte` are you using?

4.2.1

### What did you do?

Trying to get linting to work for both TypeScript files and Svelte using the new ESLint flat config. So far, I can get the ESLint config to work for either TypeScript file or Svelte files, but not both due to an issue relating to `tsconfig.ts`. I don't know whether this is a bug in my configuration, or whether it is an issue with this plugin and/or the TypeScript plugin.

While I understand the flat config is still very new and not fully released, ESLint does plan for the new configuration file to replace the existing `.eslintrc.cjs` files.

<details>
<summary><code>eslint.config.js</code></summary>

```js
import js from "@eslint/js";
import typescriptPlugin from "@typescript-eslint/eslint-plugin";
import typescriptParser from "@typescript-eslint/parser";
import sveltePlugin from "eslint-plugin-svelte";
import svelteParser from "svelte-eslint-parser";

export default [
  {
    ignores: [".svelte-kit/**/*"],
  },

  // Load predefined config
  js.configs.recommended,

  // JavaScript
  {
    files: ["**/*.js", "**/*.cjs"],
  },

  // TypeScript
  {
    files: ["**/*.ts"],
    ignores: ["playwright.config.ts"],
    languageOptions: {
      parser: typescriptParser,
      parserOptions: {
        project: "./tsconfig.json",
      },
    },
    plugins: {
      "@typescript-eslint": typescriptPlugin,
    },
    rules: {
      ...typescriptPlugin.configs.recommended.rules,
    },
  },

  // Svelte
  {
    files: ["**/*.svelte"],
    languageOptions: {
      parser: svelteParser,
      parserOptions: {
        parser: typescriptParser,
        project: "./tsconfig.json",
        extraFileExtensions: [".svelte"],
      },
    },
    plugins: {
      svelte: sveltePlugin,
      "@typescript-eslint": typescriptPlugin,
    },
    rules: {
      ...typescriptPlugin.configs.recommended.rules,
      ...sveltePlugin.configs.recommended.rules,
    },
  },
];
```
</details>

<details>
<summary><code>src/routes/+pages.svelte</code></summary>

```svelte
<!-- Welcome to eslint-plugin-svelte -->
<script lang="ts">
  let a = 1;
  let b = 2;
  // let c = 2;
  let string = `this string contains some <strong>HTML!!!</strong>`;
  let user = {
    firstname: 'Ada',
    lastname: 'Lovelace'
  };
    let current = 'foo';
  let color = 'red';
  let active = true;
</script>

<input
  type="number"
  bind:value={a}>
<input
    type="number"
  bind:value={b}>
<input
  type="number"
  bind:value={c}>
<p>{a} + {b} + {c} = {a + b + c}</p>

<p>{@html string}</p>

<input bind:value={user.firstname}>
<input bind:value={user.lastname}>

{@debug user}

<h1>Hello {user.firstname}!</h1>

{#if a}
  <div>foo</div>
{:else if b}
  <div>bar</div>
{:else if b}
  <div>baz</div>
{/if}

<button
  type=button
  class="{active ? 'active' : ''} {current === 'foo' ? 'selected' : ''}"
  on:click="{() => current = 'foo'}"
>foo</button>

<div style="color: {color};">...</div>
```

</details>

<details>
<summary><code>src/test.ts</code></summary>

```ts
let myVar = 100;
myVar = "foo";
console.log(myVar);

const myConst = 100;
myConst += 1;
console.log(myConst);

export { };
```

</details>

### What did you expect to happen?

Errors to be reported in both the TypeScript and the Svelte files.

### What actually happened?

If I enable both the Svelte and TypeScript configurations in the ESLint flat configuration, the Svelte file produces the following error (noting that the TypeScript file works fine):

```text
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/src/routes/+page.svelte` using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
```

If I disable the TypeScript section, the Svelte file produces the expected errors, but then the TypeScript file produces no lints (as it is no longer included).

### Link to **GitHub Repo** with Minimal Reproducible Example

[JP-Ellis/eslint-plugin-svelte-mwe](https://github.com/JP-Ellis/eslint-plugin-svelte-mwe)

### PR DESCRIPTION
**Pull request by JP-Ellis**
The TypeScript parser uses a singleton internally, which means that repeated `parserOptions` will be ignored, using only the first one. This can lead to unexpected behaviour if there are multiple possible configurations.

This closes #422.

Ref: #422

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Introduction
 
-`eslint-plugin-svelte` is [ESLint] plugin for [Svelte].
```
Comment (ota-meshi) [inline README.md:3]: The two spaces at the end of the line are intended to be `<br>`. Please undo the change or insert `\` instead.

**Review by ota-meshi — APPROVED** [review]
Thank you very much!

File: README.md (hunk 1 above)
Comment (ota-meshi) [reply README.md]: I recently learned that I can use backslashes in markdown.
It would be nice for me if a formatter or something could fix trailing spaces to backslashes, but I haven't found it yet 😓

Comment (JP-Ellis) [reply README.md]: You could try `sed -i "s/  $/ \\/"` maybe? You might just need to review the changes in case changes anything unexpected.
